src/shared/state/assets.ts

```diff
@@ -0,0 +1,32 @@
+import { ViewService } from '@penumbra-zone/protobuf';
+import { getDenomMetadata } from '@penumbra-zone/getters/assets-response';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { penumbra } from '@/shared/penumbra';
+import { connectionStore } from '@/shared/state/connection';
+import { useQuery } from '@tanstack/react-query';
+import { useEnv } from '@/fetchers/env';
+
+export const useAssets = () => {
+  const { data: env, isLoading } = useEnv();
+  const chainId = !isLoading ? env?.PENUMBRA_CHAIN_ID : process.env['PENUMBRA_CHAIN_ID'];
```

Comment:
thought: I didn't know the client has both client-side env variables and remote ones we fetch (that overlap). This feels a bit brittle and feels like we should really only have one or the other.

Regardless, I think the logic of choosing which env vars are correct should be in `useEnv` and not have to be dealt with by other hooks (and add a code comment about the two different env vars there).